### PR TITLE
feat: message if wrangler needs updating

### LINKED ISSUE #397: Print update message on wrangler usage when new version of wrangler is available.
To encourage wrangler users to use the latest version of the CLI, we should print out a message along the lines of 
```
A new version of wrangler (1.x.0) is available! Make sure to update by following the update instructions: https://github.com/cloudflare/wrangler#updating-wrangler
```
whenever a new version of wrangler is available. This could be printed out before every wrangler invocation if a user's wrangler version is outdated.

This version information could be pulled from github, crates.io, or npm; figuring out how to manage this without accidentally DOSing one of these endpoints is useful. (I know github rate limits calls to the API used to access repository tags, so we should keep this in mind).

### PR DESCRIPTION
**Pull request by EverlastingBugstopper**
Fixes #397

every 24 hours, wrangler will ping crates.io and ask for the latest version. if latest version is greater than currently installed version, a message will be printed stating that the version needs updating. it saves the result from crates.io at `$WRANGLER_HOME/version.toml` which includes the latest version and the time it was last checked.

### CODE REVIEW

**Comment by victoriabernard92** [comment]
Since there aren't tests yet could we add an undocumented/temp option to turn this off incase it breaks anyone? It could live in ` $WRANGLER_HOME/config/default.toml` as 
```
api_token="0xbd.."
warn_update="false"
```
We could remove this later once we know users aren't irritated by it

Just an idea. LGTM either way

File: src/util/version.rs [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+pub fn background_check_for_updates() -> mpsc::Receiver<Version> {
+    let (sender, receiver) = mpsc::channel();
+
+    let _detached_thread = thread::spawn(move || match check_wrangler_versions() {
+        Ok(wrangler_versions) => {
+            if wrangler_versions.current != wrangler_versions.latest {
+                let _ = sender.send(wrangler_versions.latest);
+            }
+        }
+        Err(e) => log::debug!("could not determine if update is needed:\n{}", e),
```
Comment (victoriabernard92) [inline src/util/version.rs:123]:
```suggestion
        Err(e) => log::debug!("could not connect to crates.io to check up to date with latest version:\n{}", e),
```

Comment (victoriabernard92) [reply src/util/version.rs:123]: Always a good idea to include a best next step for the user to take even if it's ignoring the message 
e.g.
```
ignore this message if the output of wrangler -V matches that at https://crates.io/api/v1/crates/wrangler
```

Comment (EverlastingBugstopper) [reply src/util/version.rs:123]: this log is not provided to the user unless they set `RUST_LOG=debug`

**Comment by EverlastingBugstopper** [comment]
so the only thing that needs to be done with this is making it only message once every 24 hours. right now, if your version of wrangler needs updating, it will print a message telling you to update after _every single command_. this shouldn't happen. Let's only bug folks about it once a day.

File: src/version/mod.rs [hunk 2]
```diff
@@ -0,0 +1,5 @@
+mod block;
```
Comment (ashleymichal) [inline src/version/mod.rs:1]: this module got moved into the wranglerjs module in a previous commit; we should be able to remove this and just have  `src/version.rs`

File: src/version/version.rs [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+    };
+
+    Ok(WranglerVersion {
+        current,
+        latest,
+        checked,
+    })
+}
+
+/// Reads version out of version file, is `None` if file does not exist
```
Comment (ashleymichal) [inline src/version/version.rs]:
```suggestion
/// Reads version out of version file, is `None` if file does not exist or is corrupted
```

File: src/version/wrangler_version.rs [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+#[derive(Deserialize, Debug)]
+struct CrateInformation {
+    max_version: String,
+}
+
+pub fn background_check_for_updates() -> mpsc::Receiver<Version> {
+    let (sender, receiver) = mpsc::channel();
+
+    let _detached_thread = thread::spawn(move || match check_wrangler_versions() {
+        Ok(wrangler_versions) => {
```
Comment (ashleymichal) [inline src/version/wrangler_version.rs:144]: nit: why plural?

File: src/version/wrangler_version.rs [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+            let time_since_last_checked =
+                current_time.duration_since(last_checked_version.last_checked)?;
+
+            if time_since_last_checked.as_secs() < ONE_DAY {
+                checked = true;
+            }
+            Version::parse(&last_checked_version.latest_version)?
+        }
+        // If version.toml doesn't exist, fetch latest version
+        None => get_latest_version(&current.to_string(), &version_file, current_time)?,
```
Comment (ashleymichal) [inline src/version/wrangler_version.rs:76]: this is  the only place we  call `get_latest_version`;  is that  right?

File: src/version/mod.rs [hunk 6]
```diff
@@ -0,0 +1,157 @@
+use std::fs;
```
Comment (ashleymichal) [inline src/version/mod.rs:1]: last nit: if this is only `version/mod.rs` it can just be `version.rs`

File: src/version/mod.rs [hunk 7]
```diff
@@ -0,0 +1,157 @@
...
+use std::path::PathBuf;
+use std::str::FromStr;
+use std::sync::mpsc;
+use std::thread;
+use std::time::SystemTime;
+
+use crate::settings::global_user::get_wrangler_home_dir;
```
Comment (ashleymichal) [inline src/version/mod.rs:8]: nit: this probably does not  belong  to global_user,  but idk where to  put it just now.

**Pull request merged**